### PR TITLE
added add to cart feature

### PR DESCRIPTION
**Pull request by ShreyashMahajan**
In this PR, add to cart feature is added

### CODE REVIEW

File: src/components/productCard/ProductCard.jsx [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
+import { useWishList } from '../../context/wishContext/wishContext';
 
 export const ProductCard = () => {
     const { finalFilteredData } = useFilter();
+    const { cartDispatch } = useCart();
+    const { wishListDispatch } = useWishList();
 
     return (
         <>
             {
```
Comment (JantuDeb) [inline src/components/productCard/ProductCard.jsx:14]: You can remove 
```
<> 
{}
</>
```

File: src/context/cartContext/cartContext.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { useReducer } from "react";
+import { cartReducer } from '../../reducerFunctions/cartReducer/cartReducer.jsx'
+
+const CartContext = createContext(null);
+
+
+const CartProvider = ({ children }) => {
+
+    const [cartState, cartDispatch] = useReducer(cartReducer, { cartList: [] });
+    console.log("cart list", cartState.cartList);
```
Comment (JantuDeb) [inline src/context/cartContext/cartContext.jsx:12]: You can do this in your cart page

File: src/context/cartContext/cartContext.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    const totalReducer = (prev, curr) => ({ ...prev, total: prev.total + curr.price * curr.quantity })
+    const total = cartState.cartList.reduce(totalReducer, { total: 0 });
+    console.log("total price ", total);
```
Comment (JantuDeb) [inline src/context/cartContext/cartContext.jsx:16]: You can do the same in cart page

**Review by JantuDeb — APPROVED** [review]

**Comment by SaurabhChawda** [comment]
Code Looks good to me, but in the login Page you have given extra space, You can reduce that. I liked that you have used two different contexts for add to cart and add to wishlist.

**Review by SaurabhChawda — APPROVED** [review]

File: src/context/cartContext/cartContext.jsx (hunk 3 above)
Comment (ShreyashMahajan) [reply src/context/cartContext/cartContext.jsx:16]: ok, i will do change that

File: src/components/productCard/ProductCard.jsx (hunk 1 above)
Comment (ShreyashMahajan) [reply src/components/productCard/ProductCard.jsx:14]: ok

**Pull request merged**